roles/cifmw_setup/tasks/deploy_architecture.yml

```diff
@@ -0,0 +1,269 @@
+          'automation/vars',
+          cifmw_architecture_scenario~'.yaml'
+        ) | ansible.builtin.path_join
+      }}
+
+- name: Load architecture automation file
+  tags:
+    - edpm_deploy
+  register: _automation
+  ansible.builtin.slurp:
+    path: "{{ cifmw_architecture_automation_file }}"
+
+- name: Prepare automation data
+  tags:
+    - edpm_deploy
+  vars:
+    _parsed: "{{ _automation.content | b64decode | from_yaml }}"
+  ansible.builtin.set_fact:
+    cifmw_deploy_architecture_steps: >-
+      {{ _parsed['vas'][cifmw_architecture_scenario] }}
+
+- name: Check requirements
+  tags:
+    - edpm_bootstrap
+  ansible.builtin.import_role:
+    name: kustomize_deploy
+    tasks_from: check_requirements.yml
+
+- name: Reduce OCP cluster size in architecture
+  when:
+    - groups['ocps'] | length == 1
+  ansible.builtin.import_role:
+    name: kustomize_deploy
+    tasks_from: reduce_ocp_cluster.yml
+  tags:
+    - edpm_bootstrap
+
+- name: Configure Storage Class
+  ansible.builtin.import_role:
+    name: ci_local_storage
+  when: not cifmw_use_lvms | default(false)
+  tags:
+    - storage
+    - edpm_bootstrap
+
+- name: Deploy OSP operators
+  ansible.builtin.import_role:
+    name: kustomize_deploy
+    tasks_from: install_operators.yml
+  tags:
+    - operator
+    - edpm_bootstrap
+
+- name: Update containers in deployed OSP operators
+  vars:
+    cifmw_update_containers_metadata: controlplane
+  ansible.builtin.include_role:
+    name: update_containers
+  tags:
+    - update_containers
+    - edpm_bootstrap
+  when: cifmw_ci_gen_kustomize_values_deployment_version is not defined
+
+- name: Update containers in deployed OSP operators using set_openstack_containers role
+  when:
+    - cifmw_set_openstack_containers | default(false) | bool
+    - cifmw_ci_gen_kustomize_values_deployment_version is not defined
+  ansible.builtin.include_role:
+    name: set_openstack_containers
+  tags:
+    - set_openstack_containers
+    - edpm_bootstrap
+
+- name: Configure LVMS Storage Class
+  ansible.builtin.include_role:
+    name: ci_lvms_storage
+  when: cifmw_use_lvms | default(false)
+  tags:
+    - storage
+    - edpm_bootstrap
+
+- name: Execute deployment steps
+  tags:
+    - edpm_deploy
+  ansible.builtin.include_role:
+    name: kustomize_deploy
+    tasks_from: execute_step.yml
+    apply:
+      tags:
+        - edpm_deploy
+  loop: "{{ cifmw_deploy_architecture_steps.stages }}"
+  loop_control:
+    label: "{{ stage.path }}"
+    loop_var: stage
+    index_var: stage_id
+
+- name: Extract and install OpenStackControlplane CA
+  ansible.builtin.import_role:
+    role: install_openstack_ca
+  tags:
+    - openstack_ca
+    - edpm_post
+
+- name: Run nova host discover process
+  tags:
+    - edpm_post
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.command: >-
+    oc rsh
+    -n openstack
```

Comment:
@abays @danpawlik should  `namespace` be a variable for multi namespace execution?

Reply:
good point. it seems we have similar bug/feature request in other tasks in the role. Would improve in  https://github.com/openstack-k8s-operators/ci-framework/pull/3098